torch/utils/cpp_extension.py

```diff
@@ -284,25 +284,27 @@ def _join_sycl_home(*paths) -> str:
 
 _COMMON_SYCL_FLAGS = [
     '-fsycl',
-    '-fsycl-targets=spir64_gen,spir64',
 ]
 
-def _get_sycl_arch_list():
-    if 'TORCH_XPU_ARCH_LIST' in os.environ:
-        return os.environ.get('TORCH_XPU_ARCH_LIST')
+def _get_sycl_arch_flags():
     arch_list = torch.xpu.get_arch_list()
     # Dropping dg2* archs since they lack hardware support for fp64 and require
     # special consideration from the user. If needed these platforms can
     # be requested thru TORCH_XPU_ARCH_LIST environment variable.
     arch_list = [x for x in arch_list if not x.startswith('dg2')]
-    return ','.join(arch_list)
+    if len(arch_list) == 0:
+        return []
```

Comment:
What the behavior of built extension will be if we return empty `[]` arch flags list?
1. Will it actually be buildable?
2. For which arch(s) it will be built?
3. Or it won't be pre-built for any AOT target and running extension will result in runtime compilation?

Secondly, is `'-fsycl-targets=spir64_gen,spir64'` still needed to be passed here?

I think empty arch list worths a comment left in the source code here.

Reply:
Compilation will still work, but without aot compilation with `ocloc`. `-fsycl-targets=spir64_gen,spir64` cannot be here, otherwise `ocloc` will crash complaining no targets are set.